components/eamxx/src/share/field/field_request.hpp

```diff
@@ -271,6 +271,7 @@ struct FieldRequest {
   SubviewInfo               subview_info;
   std::string               parent_name;
   bool                      incomplete = false;
+  std::string               calling_process = "";
```

Comment:
A bit hacky, but wanted the error message in the AD to be able to list the processes and what they request w.r.t. SHOC advected. We could opt for a much simpler check and message "Error! qv has incompatible requests for turbulence advection" or something like that.

Or this can become more integrated in FieldRequest and there could be a dedicated constructor arg.

Reply:
I think this is perfectly fine. Maybe default to "UNKNOWN" or something like that?